lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -222,13 +267,86 @@ class TypeAnalyzer : public ForwardDataFlowAnalysis<ValueKnowledge> {
     } else if (auto anyDim = dyn_cast<AtenAnyDimOp>(op)) {
       return visitAtenAnyDimOp(anyDim, operands);
     } else if (auto view = dyn_cast<AtenViewOp>(op)) {
-      return visitAtenViewOp(view, operands);
+      return visitReshapeLikeOp(view, operands);
+    } else if (auto resize = dyn_cast<AtenResize_Op>(op)) {
+      return visitReshapeLikeOp(resize, operands);
     } else if (auto transposeInt = dyn_cast<AtenTransposeIntOp>(op)) {
       return visitAtenTransposeIntOp(transposeInt, operands);
+    } else if (auto tensorFloat = dyn_cast<AtenTensorFloatOp>(op)) {
+      return visitScalarToTensorConversionOp<AtenTensorFloatOp>(tensorFloat);
+    } else if (auto tensorInt = dyn_cast<AtenTensorIntOp>(op)) {
+      return visitScalarToTensorConversionOp<AtenTensorIntOp>(tensorInt);
+    } else if (auto tensorBool = dyn_cast<AtenTensorBoolOp>(op)) {
+      return visitScalarToTensorConversionOp<AtenTensorBoolOp>(tensorBool);
+    } else if (auto tensor = dyn_cast<AtenTensorOp>(op)) {
+      return visitAtenTensorOp(tensor);
+    } else if (auto zeros = dyn_cast<AtenZerosOp>(op)) {
+      return visitConstantTensorAllocOp<AtenZerosOp>(zeros);
+    } else if (auto ones = dyn_cast<AtenOnesOp>(op)) {
+      return visitConstantTensorAllocOp<AtenOnesOp>(ones);
+    } else if (auto emptyMemoryFormat = dyn_cast<AtenEmptyMemoryFormatOp>(op)) {
+      return visitConstantTensorAllocOp<AtenEmptyMemoryFormatOp>(
+          emptyMemoryFormat);
+    } else if (auto toOther = dyn_cast<AtenToOtherOp>(op)) {
+      return visitTypeConversionOp<AtenToOtherOp>(toOther, operands);
+    } else if (auto typeAs = dyn_cast<AtenTypeAsOp>(op)) {
+      return visitTypeConversionOp<AtenTypeAsOp>(typeAs, operands);
+    } else if (auto indexSelect = dyn_cast<AtenIndexSelectOp>(op)) {
+      // The index tensor index into the dimension specified by the dim. The dim
+      // of output is the same size as the length of index (index must be one
+      // dimensional)
+      auto setDim = [](int64_t &targetDim, int64_t dim,
+                       ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+        auto indexes = operands[2]->getValue();
+        targetDim = indexes.hasSizes ? indexes.sizes[0] : kUnknownSize;
```

Comment:
need to check `indexes.sizes.size() != 0` -- the user pass a 0d tensor for `indexes` (it could be hidden behind an unreachable branch though -- we cannot crash the compiler).